app/services/flowcommerce_spree/webhooks/capture_upserted_v2.rb

```diff
@@ -22,16 +22,25 @@ def process
         errors << { message: 'Order number param missing' } && (return self) unless order_number
 
         if (order = Spree::Order.find_by(number: order_number))
-          upsert_order_captures(order, capture)
-          payments = order.flow_io_payments
-          map_payment_captures_to_spree(order, payments) if payments.present?
-          order
+          if order.payments.any?
+            store_payment_capture(order, capture)
+          else
+            FlowcommerceSpree::UpdatePaymentCaptureWorker.perform_in(1.minute, order.number, capture)
```

Comment:
this minute is the ultimate time? this time will always be enough?

Reply:
Sorry @maurosettier for the delay!! Yes, it should be enough! This method generally does not take more than seconds. If there is a problem though (timeout, bad gateway. etc) We would end up getting informed in the Slack channel `#flow-io-alerts` thanks to how we set up the DelayedJob!